packages/react-component-library/src/components/Pagination/Pagination.test.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import { render, fireEvent, RenderResult } from '@testing-library/react'
+
+import Pagination from './index'
+
+describe('Pagination', () => {
+  let pagination: RenderResult
+  let onChangeCallback: () => void
+
+  describe('when the Pagination is generated with pageSize, total and onChangeCallback props', () => {
+    onChangeCallback = jest.fn()
+
+    beforeEach(() => {
+      pagination = render(
+        <Pagination
+          onChangeCallback={onChangeCallback}
+          pageSize={10}
+          total={1000}
+        />
+      )
+    })
+
+    it('should output the correct number of pages', () => {
```

Comment:
I'd add some tests for when there's less total pages than could be shown (i.e. there's only 2 results and you can show 10 and when there's only 20 results to make sure it doesn't show page number 3 etc